src/main/java/com/google/api/codegen/util/NamePath.java

```diff
@@ -75,6 +76,24 @@ public NamePath append(String head) {
   }
 
   /**
+   * Create a new NamePath where each name piece now starts with an uppercase word.
+   */
+  public NamePath withUpperPieces() {
+    List<String> newPathPieces = new ArrayList<>();
+    for (String piece : pathPieces) {
+      String newPiece =
+          (piece.isEmpty())
+              ? piece
+              : new StringBuilder(piece.length())
+                  .append(Ascii.toUpperCase(piece.charAt(0)))
+                  .append(piece.substring(1))
+                  .toString();
+      newPathPieces.add(newPiece);
+    }
+    return new NamePath(newPathPieces);
+  }
```

Comment:
I have concluded that until we can use Java 8 and passing functions around is easier, we will just have to live with this pattern.